internal/ec_errors/ec_error.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package ec_errors
+
+type Error struct {
```

Comment:
It would help me understand this more if this could be incorporated in a couple of places in the code base.

Reply:
Something like:

```go
if err != nil {
    return errors.GE001.CausedBy(err)
}
```

Though the first followup PR should remove `GE001`, and go through all those `if err != nil` and assign them an Error. I don't think we want any general errors, we want specific errors we can trace back to code.

Reply:
Thanks! I'm more interested in where in the code base this will done. See my [comment](https://github.com/hacbs-contract/ec-cli/pull/123#discussion_r933531712) about scope.